dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitLabSourceCodeRepo.java

```diff
@@ -159,26 +161,40 @@ public Workflow setupWorkflowVersions(String repositoryId, Workflow workflow, Op
 
     @SuppressWarnings("checkstyle:parameternumber")
     private void handleVersionOfWorkflow(String repositoryId, Workflow workflow, Optional<Workflow> existingWorkflow,
-        Map<String, WorkflowVersion> existingDefaults, String id, String branchName, Version.ReferenceType type, Date committedDate, String commitId) {
-        // Initialize workflow version
-        WorkflowVersion version = initializeWorkflowVersion(branchName, existingWorkflow, existingDefaults);
-        String calculatedPath = version.getWorkflowPath();
-
-        // Now grab source files
-        DescriptorLanguage.FileType identifiedType = workflow.getFileType();
-        // TODO: No exceptions are caught here in the event of a failed call
-        SourceFile sourceFile = getSourceFile(calculatedPath, id, branchName, identifiedType);
-
-        version.setReferenceType(type);
-        version.setLastModified(committedDate);
-        version.setCommitID(commitId);
-        // Use default test parameter file if either new version or existing version that hasn't been edited
-        createTestParameterFiles(workflow, id, branchName, version, identifiedType);
-        version = combineVersionAndSourcefile(repositoryId, sourceFile, workflow, identifiedType, version, existingDefaults);
-
-        version = versionValidation(version, workflow, calculatedPath);
-
-        workflow.addWorkflowVersion(version);
+        Map<String, WorkflowVersion> existingDefaults, String id, String branchName, Version.ReferenceType type, Date committedDate, String commitId, boolean hardRefresh) {
+        if (toRefreshVersion(branchName, commitId, existingDefaults, hardRefresh)) {
```

Comment:
Yeah there are some areas that I didn't want to mess with too much since it is a hotfix. Since this is going into a later hotfix though, we will have more time to test. So I think we could get away with bigger changes.